### PR TITLE
Use global package reference

### PR DESCRIPTION
**Pull request by WeihanLi**
- use `GlobalPackageReference` for `StyleCop.Analyzers`
- remove `Microsoft.SourceLink.GitHub` since the source link and this package had been included in .NET 8 SDK

https://learn.microsoft.com/en-us/nuget/consume-packages/central-package-management#global-package-references

https://github.com/dotnet/sourcelink?tab=readme-ov-file#using-source-link-in-net-projects

### CODE REVIEW

File: src/Directory.Packages.props [hunk 1]
```diff
@@ -66,4 +64,7 @@
     <PackageVersion Include="Microsoft.Extensions.Configuration.Binder" Version="6.0.0" />
     <PackageVersion Include="Npgsql" Version="7.0.7" />
   </ItemGroup>
+  <ItemGroup>
+    <GlobalPakcgaeReference Include="StyleCop.Analyzers" Version="1.1.118" />
```
Comment (seantleonard) [inline src/Directory.Packages.props:68]:
```suggestion
    <GlobalPackageReference Include="StyleCop.Analyzers" Version="1.1.118" />
```

**Review by Aniruddh25 — APPROVED** [review]
Thank you for your contributions!!

**Comment by WeihanLi** [comment]
@seantleonard sure, updated, thanks

**Review by seantleonard — APPROVED** [review]

**Comment by WeihanLi** [comment]
@seantleonard could you please help on the failed CI, think not related to the changes, maybe we could retry the failed CI job?

**Comment by seantleonard** [comment]
Thank you for your contribution, @WeihanLi !

**Pull request merged**